### PR TITLE
fix(processor): bulletproof ortho nodata detection at image_reprojector

### PR DESCRIPTION
**Pull request by cmosig**
## Problem

The embedding explorer (and segmentation) treated clearly-nodata areas as real imagery — e.g. [dataset 3789](https://deadtrees.earth/dataset/3789), whose white border is nodata but got embedded/segmented anyway.

Root cause is a shared chokepoint: every stage that reprojects an ortho (embeddings + all three segmentation inferences) goes through `image_reprojector`, which forces `nodata=0` on the `WarpedVRT`. That masks black fill and the warp border, but silently misses two common conventions:

1. **A mislabeled extra band** — an RGB ortho with a 4th band that is really a 0/255 alpha but tagged `undefined`/`gray`, so nothing treats it as transparency (3789's exact shape).
2. **Solid white fill** on orthos that carry *no* nodata metadata at all.

Prod scan shows this isn't a one-off: ~319 datasets declare `nodata=255`, ~220 have a mislabeled 4th band, ~1,184 are RGB with no mask/nodata.

## Fix

Resolve a `NodataPolicy` **once** from the source in `image_reprojector`, attach it to the VRT, and have every consumer read the mask through a single shared `read_nodata_mask()` (new `processor/src/utils/nodata.py`).

Verified empirically that the forced-`nodata=0` warp already makes `read_masks` reflect a **real alpha band, an internal mask, and a declared nodata value** (0/255/NaN/65535/…) plus the warp border. So the policy only adds the two things GDAL can't infer on its own:

- **Mislabeled binary mask band** — detected by sampling (near-binary, contains zeros, not a colour band; continuous NIR is never mistaken for a mask).
- **White/black fill fallback** — enabled **only** when the source declares no masking whatsoever, so the 6,000+ well-formed datasets are untouched and take the same `read_masks` path as before (no perf change).

Also tightens the embedding-explorer tile filter to require **>99% real data** (`NODATA_THRESHOLD` 0.5 → 0.01).

## Consumers migrated
`read_masks(1) == 0` → `read_nodata_mask(vrt, window)` in:
- `embedding_search/patch_embedding.py`
- `deadwood_treecover_combined_v2` / `deadwood_segmentation_v1_moehring` / `aoi_segmentation_v1` inferences

## Tests
New `processor/tests/test_nodata.py` (23 tests): resolver branch-per-convention, `_detect_mask_band` edge cases, `read_nodata_mask` combination logic, plus **end-to-end through the real `image_reprojector`** (alpha band, declared white nodata, internal mask, un-tagged white fill, clean-imagery-masks-nothing). All green.

## ⚠️ Takes effect on reprocessing
Existing rows were produced with the old logic. Affected datasets (3789, the `nodata=255` set, the mislabeled-4th-band set, no-metadata fill orthos) need their **embeddings and segmentation re-run** to benefit. No reprocessing is queued by this PR.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: processor/src/utils/nodata.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	mask = vrt.read_masks(1, window=window) == 0
+
+	if policy.mask_band is not None and policy.mask_band <= vrt.count:
+		mask = mask | (vrt.read(policy.mask_band, window=window) == 0)
+
+	if policy.treat_white_fill or policy.treat_black_fill:
+		n = min(3, vrt.count)
+		rgb = vrt.read(indexes=list(range(1, n + 1)), window=window)
+		if policy.treat_white_fill:
+			mask = mask | np.all(rgb == 255, axis=0)
```
[the remark this answers is not in the document]
Comment (cmosig) [reply processor/src/utils/nodata.py]: Good catch — fixed in caca63e. The white/black fill fallback now only masks solid fill that is 4-connectivity-**connected to already-known nodata or a true raster edge** (i.e. actual footprint padding), via a pure-numpy border-flood. Interior saturated-white content (snow, panels, sunlit roofs) isn't border-connected and is kept. Added tests covering an interior white patch being preserved while edge-connected padding is masked, both at the unit level and end-to-end through `image_reprojector`.
